internal/store/chain.go

```diff
@@ -15,30 +14,20 @@ import (
 var (
 	ErrBlockNotFound  = errors.New("block not found")
 	ErrHeaderNotFound = errors.New("header not found")
-	ErrChainClosed    = errors.New("chain store is closed")
-)
-
-const (
-	prefixHeader byte = iota + 1
-	prefixBlock
 )
 
 // Chain manages blockchain storage using a key-value store
 type Chain struct {
-	db     db.KVStore
-	closed atomic.Bool
+	db.KVStore
```

Comment:
What's the advantage of embedding here? (embedding tends to have a lot more footguns to be aware of)